moveit_planners/chomp/chomp_motion_planner/src/chomp_optimizer.cpp

```diff
@@ -388,19 +352,9 @@ void ChompOptimizer::optimize()
     // ROS_INFO_STREAM("Collision increments took " << (ros::WallTime::now()-coll_time));
     calculateTotalIncrements();
 
-    // if(!parameters_->getUseHamiltonianMonteCarlo())
```

Comment:
So Hamiltonian Monte Carlo is an important part of maintaining CHOMP's completeness/resilience to local minimums. It does have trade offs and I understand that you might not have time to verify and test this feature as much, but IMO this should be kept, or at least marked as a priority for the next time CHOMP is worked on.  Maybe the default should be set to false, and the feature can be documented as experimental?

Reply:
Yes I agree to what you say and this also well aligns with the research paper. However none of the HMC based code was getting used and it was all commented. I did try and uncomment all the HMC part for CHOMP to avoid local minima while avoiding obstacles. It was rarely able to break out of the local minima. 

I could do some more extensive testing later in July . Meanwhile I could leave this part commented and mark this feature documented as experimental if required.

Reply:
I don't know the background theory to this but, based on Bryce's feedback, I'd suggest we keep the commented out stuff around Hamiltonian Monte Carlo but add a TODO note explaining why it is commented out, why it is important, and what might need to be done to restore it / fix it